crates/vm/levm/src/db/gen_db.rs

```diff
@@ -47,12 +53,41 @@ impl GeneralizedDatabase {
         &mut self,
         accrued_substate: &mut Substate,
         address: Address,
-    ) -> Result<(&Account, bool), DatabaseError> {
+    ) -> Result<(&Account, bool), VMError> {
         let address_was_cold = accrued_substate.touched_accounts.insert(address);
         let account = self.get_account(address)?;
 
         Ok((account, address_was_cold))
     }
+
+    /// Gets account from storage, storing in Immutable Cache for efficiency when getting AccountUpdates.
+    pub fn get_account_from_database(&mut self, address: Address) -> Result<Account, VMError> {
+        let account = self.store.get_account(address)?;
+        self.immutable_cache.insert(address, account.clone());
+        Ok(account)
+    }
+
+    /// Gets storage slot from Database, storing in Immutable Cache for efficiency when getting AccountUpdates.
+    pub fn get_value_from_database(
+        &mut self,
+        address: Address,
+        key: H256,
+    ) -> Result<U256, VMError> {
+        let value = self.store.get_storage_value(address, key)?;
```

Comment:
Same here, can't we make a lookup to the immutable cache first in case we find it?